data/lua/core/map.lua

```diff
@@ -461,6 +461,9 @@ if wesnoth.kernel_type() == "Mapgen Lua Kernel" then
 	wesnoth.map.get_tiles_radius = wesnoth.deprecate_api('map:get_tiles_radius', 'map:find_in_radius', 1, nil, function(map, locs, filter, radius)
 		return wesnoth.map.find_in_radius(map, locs, radius, filter)
 	end, 'The filter is now the last parameter, instead of the radius')
+	wesnoth.map.vector_sum = wesnoth.deprecate_api('wesnoth.map.vector_sum', 'wesnoth.map.hex_vector_sum', 1, nil, wesnoth.map.hex_vector_sum)
+	wesnoth.map.vector_diff = wesnoth.deprecate_api('wesnoth.map.vector_diff', 'wesnoth.map.hex_vector_dif', 1, nil, wesnoth.map.hex_vector_diff)
+	wesnoth.map.vector_negation = wesnoth.deprecate_api('wesnoth.map.vector_negation', 'wesnoth.map.hex_vector_negation', 1, nil, wesnoth.map.hex_vector_negation)
```

Comment:
Given that the old functions were already in `map`, it was obvious that they were something to do with the map. What's the advantage of renaming them with `hex_` ?

Reply:
I thought the old names gave the impression that they represented simple vector arithmetic, like `(x1, y1) + (x2, y2) = (x1 + y1, x2 + y2)`.

Reply:
I assumed that these were using the same coordinates as the `hex_locations`, but it seems they're using x,y instead. So the new names with `hex_` in are worst than the old ones, even without the cost of deprecation.

Reply:
I'm open to better suggestions, either for the functions or for `hex_location`. The key issue I have is that someone might see "vector" and think it's working in Cartesian space, but it's actually using hex-offset coordinates (the coordinates used by `map_location`). The name `hex_location` is _also_ perhaps not the best, as both `hex_location` and `map_location` are hex-based coordinates, just _two different_ types of hex-based coordinates. The `hex_location` version makes some algorithms easier, while the `map_location` version is better for addressing maps as a pseudo-rectangular grid.

Reply:
Leaving it as-is until a better name comes up is better than changing the name twice. I think the current name is the best option for now.